cluster/cluster.go

```diff
@@ -252,3 +253,32 @@ func (c *Cluster) getUniqueHostList() []*hosts.Host {
 	}
 	return uniqHostList
 }
+
+func (c *Cluster) DeployAddons(ctx context.Context) error {
+	if err := c.DeployK8sAddOns(ctx); err != nil {
+		return err
+	}
+	if err := c.DeployK8sIngress(ctx); err != nil {
```

Comment:
Is there a reason why ingress is treated specially here, and not a part of DeployK8sAddOns (like dns)? it is really similar, and probably should be a part of k8s addons